Better metric descriptions and readme docs (#19933)
DataDog/integrations-core: infiniband/README.md

Key metrics collected include port counters like bytes/packets transmitted and received, error counts, and RDMA hardware-specific metrics - giving operators the data needed to ensure optimal performance of their high-speed networking infrastructure.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. The check collects metrics by reading and submitting counters by default from [`/sys/class/infiniband/<device>/ports/*/counters/` and `/sys/class/infiniband/<device>/ports/*/hw_counters/`][3] directories. To ensure that this integration works, ensure that the Agent has the appropriate permissions to access and read the counters from these directories.

### Installation

The InfiniBand check is included in the [Datadog Agent][2] package.
No additional installation is needed on your server.

### Configuration

> 1. Create and edit the `infiniband.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your infiniband performance data. See the [sample infiniband.d/conf.yaml][4] for all available configuration options.

Reviewer: ```suggestion
1. To start collecting your InfiniBand performance data, create and edit the `infiniband.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory. See the [sample infiniband.d/conf.yaml][4] for all available configuration options.
```

> 2. This check works with minimal configuration. Optional parameters are provided to better control where the Agent looks for data and what data to collect if the default behaviors are not desired. Options include configuring the directory where counters reside, excluding specific devices/ports, and skipping or adding counters for collection.

Reviewer: ```suggestion
2. This check works with minimal configuration. Configure optional parameters, which are provided to better control where the Agent looks for data and what data to collect if the default behaviors are not desired. Options include configuring the directory where counters reside, excluding specific devices/ports, and skipping or adding counters for collection.
```
Because this is a step, I wanted to add in a verb here to make it clearer what the user is supposed to do.

```yaml
init_config:
instances:
  -
    ## @param infiniband_path - string - optional - default: /sys/class/infiniband
    ## The path to the infiniband directory.
    #
    # infiniband_path: /sys/class/infiniband

    ## @param exclude_devices - list of strings - optional
    ## A list of devices to exclude from the check. Devices are located in the infiniband directory. 

>     ## The devices are located default is /sys/class/infiniband.

Reviewer: ```suggestion
    ## The devices are located by default in /sys/class/infiniband.
```
